apps/admin-portal/src/store/actions/authenticate.ts

```diff
@@ -0,0 +1,245 @@
+                            "views:components.profile.notifications.getProfileInfo.error.description",
+                            { description: error.response.data.detail }
+                        ),
+                        level: AlertLevels.ERROR,
+                        message: i18n.t(
+                            "views:components.profile.notifications.getProfileInfo.error.message"
+                        )
+                    })
+                );
+
+                return;
+            }
+
+            dispatch(
+                addAlert({
+                    description: i18n.t(
+                        "views:components.profile.notifications.getProfileInfo.genericError.description"
+                    ),
+                    level: AlertLevels.ERROR,
+                    message: i18n.t(
+                        "views:components.profile.notifications.getProfileInfo.genericError.message"
+                    )
+                })
+            );
+        })
+        .finally(() => {
+            dispatch(setProfileInfoLoader(false));
+        });
+};
+
+/**
+ * Handle user sign-in
+ */
+export const handleSignIn = (consentDenied: boolean= false) => (dispatch) => {
+    const sendSignInRequest = () => {
+        const requestParams: OIDCRequestParamsInterface = {
+            clientHost: GlobalConfig.clientHost,
+            clientId: GlobalConfig.clientID,
+            clientSecret: null,
+            enablePKCE: true,
+            redirectUri: GlobalConfig.loginCallbackUrl,
+            scope: [TokenConstants.LOGIN_SCOPE, TokenConstants.INTERNAL_IDENTITY_MGT.INTERNAL_IDENTITY_MGT_VIEW,
+                TokenConstants.INTERNAL_IDENTITY_MGT.INTERNAL_IDENTITY_MGT_UPDATE,
+                TokenConstants.INTERNAL_IDENTITY_MGT.INTERNAL_IDENTITY_MGT_DELETE,
+                TokenConstants.INTERNAL_IDENTITY_MGT.INTERNAL_IDENTITY_MGT_CREATE,
+                TokenConstants.INTERNAL_USER_MGT.INTERNAL_USER_MGT_VIEW,
+                TokenConstants.INTERNAL_USER_MGT.INTERNAL_USER_MGT_UPDATE,
+                TokenConstants.INTERNAL_USER_MGT.INTERNAL_USER_MGT_DELETE,
+                TokenConstants.INTERNAL_USER_MGT.INTERNAL_USER_MGT_LIST,
+                TokenConstants.INTERNAL_USER_MGT.INTERNAL_USER_MGT_CREATE
+            ],
+            serverOrigin: GlobalConfig.serverOrigin
+        };
+
+        if (consentDenied) {
+            requestParams.prompt = "login";
+        }
+
+        if (SignInUtil.hasAuthorizationCode()) {
+            SignInUtil.sendTokenRequest(requestParams)
+                .then((response) => {
+                    AuthenticateSessionUtil.initUserSession(
+                        response,
+                        SignInUtil.getAuthenticatedUser(response.idToken)
+                    );
+                    dispatch(setSignIn());
+                    dispatch(getProfileInformation());
+                })
+                .catch((error) => {
+                    throw error;
```

Comment:
IMO this error should be shown in an error page as the other errors. WDYT?

Reply:
The error thrown here will be handled by the error handler in the Http module.